README.md

```diff
@@ -467,3 +468,39 @@ They are as follows:
 [flipcoin code example]: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/docs/example_pipelines/condition.py
 [installodh]: https://opendatahub.io/docs/quick-installation
 [kfp-tekton]: https://github.com/kubeflow/kfp-tekton
+
+# Deployment and Testing Guidelines for Developers
+
+To run the DSPO locally :
+
+Clone the directory and execute the following command:
+
+```bash
+go run main.go --config=$config.yaml
+```
+Below is the sample config file(the tags for the images can be edited as required):
+```bash
+Images:
+  ApiServer: quay.io/opendatahub/ds-pipelines-api-server:main-0e8a011
+  Artifact: quay.io/opendatahub/ds-pipelines-artifact-manager:main-0e8a011
+  OAuthProxy: registry.redhat.io/openshift4/ose-oauth-proxy:v4.12.0
+  PersistentAgent: quay.io/opendatahub/ds-pipelines-persistenceagent:main-0e8a011
+  ScheduledWorkflow: quay.io/opendatahub/ds-pipelines-scheduledworkflow:main-0e8a011
+  Cache: registry.access.redhat.com/ubi8/ubi-minimal
+  MoveResultsImage: registry.access.redhat.com/ubi8/ubi-micro
+  MariaDB: registry.redhat.io/rhel8/mariadb-103:1-188
+  MlmdEnvoy: quay.io/opendatahub/ds-pipelines-metadata-envoy:1.7.0
+  MlmdGRPC: quay.io/opendatahub/ds-pipelines-metadata-grpc:1.0.0
+  MlmdWriter: quay.io/opendatahub/ds-pipelines-metadata-writer:1.1.0
+  CRDViewer: gcr.io/ml-pipeline/viewer-crd-controller:2.0.0-rc.2
+  VisualizationServer: gcr.io/ml-pipeline/visualization-server:2.0.0-rc.2
+```
+To run the tests:
+ 
+Execute `make test` or `make unittest` or `make functest` based on the level of testing that needs to be done.
```

Comment:
Let's expand on what each of these do.  example: functests are functional tests (what does that mean?), `make test` just runs both unittests and functests, etc